flooder/src/main.rs

```diff
@@ -1,100 +1,277 @@
-                };
-
-                for i in 0..round_count {
-                    info!("starting round #{}", i);
-                    let start = time::Instant::now();
-
-                    info!("sending #{} transactions", tx_count);
-                    for _ in 0..tx_count {
-                        conn.transfer_keep_alive(dest.clone(), transfer_amount, status)
-                            .await
-                            .unwrap();
-                    }
+    pool_limit: u64,
+) -> anyhow::Result<Vec<Nonce>> {
+    let start = Instant::now();
+    let start_finalized_hash = connections[0].get_finalized_block_hash().await?;
+    let start_finalized_number = connections[0]
+        .get_block_number(start_finalized_hash)
+        .await?
+        .unwrap() as u64;
+
+    let mut nonces: Vec<u32> = vec![];
+    for conn in connections {
+        nonces.push(conn.account_nonce(conn.account_id()).await?);
+    }
+
+    let mut overdue_transactions = 0;
+    for i in 0..schedule.intervals {
+        let pending_in_pool = connections[0].pending_extrinsics_len().await?;
+        overdue_transactions += schedule.transactions_in_interval;
+        let transactions_to_pool_limit = pool_limit.saturating_sub(pending_in_pool);
+        let transactions_to_send = min(transactions_to_pool_limit, overdue_transactions);
+        overdue_transactions -= transactions_to_send;
+        info!(
+            "Interval {}: submitting {} txns",
+            i + 1,
+            transactions_to_send
+        );
+        debug!(
+            "In the txpool there are {pending_in_pool} txns. Overdue txns: {overdue_transactions}"
+        );
+
+        let tasks = connections
+            .iter()
+            .enumerate()
+            .map(|(conn_id, conn)| {
+                let mut per_task = transactions_to_send / connections.len() as u64;
+                if (conn_id as u64) < transactions_to_send % connections.len() as u64 {
+                    per_task += 1;
+                }
 
-                    let dur = time::Instant::now().saturating_duration_since(start);
+                let dest = dest.clone();
+                let nonce = nonces[conn_id];
+                nonces[conn_id] += per_task as u32;
 
-                    let left_duration = Duration::from_secs(time).saturating_sub(dur);
+                // Set mortality roughly to flooding length
+                let params = BaseExtrinsicParamsBuilder::default().era(
+                    Era::mortal(
+                        schedule.intervals * schedule.interval_duration + 30,
+                        start_finalized_number,
+                    ),
+                    start_finalized_hash,
+                );
 
-                    info!("sleeping for {}ms", left_duration.as_millis());
-                    sleep(left_duration).await;
+                async move {
+                    for tx_id in 0..per_task as u32 {
+                        conn.sign_with_params(
```

Comment:
I wonder how much time do we spend on generating and signing txs and how much on submitting them.